user-manager/src/main/java/com/qpay/usermanager/service/impl/MerchantServiceImpl.java

```diff
@@ -40,29 +52,47 @@ public MerchantEntity getMerchantById(final long id) {
     @Override
     public MerchantEntity addMerchant(final MerchantCreation merchantCreation) {
         existsCustomerByEmail(merchantCreation.email());
```

Comment:
Maybe I don't understand smth. Why during merchant creation you check customer, but for customer you check merchant ?
At least your tests have to check it. Review your tests and make sure they check not only positive cases